### PR TITLE
Migrate criteria filters to expert filters

### PR DESCRIPTION
**Pull request by basseche**
migration all criteria based filters to expert filters, code clean will be done in the next PR

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,6 +45,7 @@
         <gridsuite-dependencies.version>35</gridsuite-dependencies.version>
         <string-template.version>4.3.1</string-template.version>
         <liquibase-hibernate-package>org.gridsuite.filter.server</liquibase-hibernate-package>
+        <sonar.coverage.exclusions>**/migrations/**/*</sonar.coverage.exclusions>
```
Comment (etiennehomer) [inline pom.xml:48]: We could try to find another solution for the location of the migration scripts. Not in this PR if it's not so easy

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 2]
```diff
@@ -65,41 +65,17 @@ public class FilterService {
     private final FilterService self;
 
     public FilterService(final ScriptFilterRepository scriptFiltersRepository,
-                         final LineFilterRepository lineFilterRepository,
-                         final GeneratorFilterRepository generatorFilterRepository,
-                         final LoadFilterRepository loadFilterRepository,
-                         final ShuntCompensatorFilterRepository shuntCompensatorFilterRepository,
-                         final StaticVarCompensatorFilterRepository staticVarCompensatorFilterRepository,
-                         final BatteryFilterRepository batteryFilterRepository,
                          final BusBarSectionFilterRepository busBarSectionFilterRepository,
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:68]: to be clean ?

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 3]
```diff
@@ -65,41 +65,17 @@ public class FilterService {
...
-                         final DanglingLineFilterRepository danglingLineFilterRepository,
                          final LccConverterStationFilterRepository lccConverterStationFilterRepository,
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:69]: to be clean

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 4]
```diff
@@ -65,41 +65,17 @@ public class FilterService {
...
                          final VscConverterStationFilterRepository vscConverterStationFilterRepository,
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:70]: to be clean ?

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 5]
```diff
@@ -65,41 +65,17 @@ public class FilterService {
...
                          NetworkStoreService networkStoreService,
                          NotificationService notificationService,
                          @Lazy FilterService self) {
-        filterRepositories.put(EquipmentType.LINE.name(), new LineFilterRepositoryProxy(lineFilterRepository));
-        filterRepositories.put(EquipmentType.GENERATOR.name(), new GeneratorFilterRepositoryProxy(generatorFilterRepository));
-        filterRepositories.put(EquipmentType.LOAD.name(), new LoadFilterRepositoryProxy(loadFilterRepository));
-        filterRepositories.put(EquipmentType.SHUNT_COMPENSATOR.name(), new ShuntCompensatorFilterRepositoryProxy(shuntCompensatorFilterRepository));
-        filterRepositories.put(EquipmentType.STATIC_VAR_COMPENSATOR.name(), new StaticVarCompensatorFilterRepositoryProxy(staticVarCompensatorFilterRepository));
-        filterRepositories.put(EquipmentType.BATTERY.name(), new BatteryFilterRepositoryProxy(batteryFilterRepository));
         filterRepositories.put(EquipmentType.BUSBAR_SECTION.name(), new BusBarSectionFilterRepositoryProxy(busBarSectionFilterRepository));
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:76]: same

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 6]
```diff
@@ -65,41 +65,17 @@ public class FilterService {
     private final FilterService self;
 
     public FilterService(final ScriptFilterRepository scriptFiltersRepository,
-                         final LineFilterRepository lineFilterRepository,
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:68]: It's a pity to change just a part of the code. We should delete all the code relative to criteria filters

File: src/main/java/org/gridsuite/filter/server/migrations/MigrateEquipmentFilter.java [hunk 7]
```diff
@@ -0,0 +1,541 @@
...
+import liquibase.exception.DatabaseException;
+import liquibase.exception.SetupException;
+import liquibase.exception.ValidationErrors;
+import liquibase.resource.ResourceAccessor;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.InsertStatement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MigrateEquipmentFilter implements CustomSqlChange {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/filter/server/migrations/MigrateEquipmentFilter.java:23]:
```suggestion
public class MigrateEquipmentFiltersToExpertFilters implements CustomSqlChange {
```

File: src/main/java/org/gridsuite/filter/server/migrations/MigrateEquipmentFilter.java [hunk 8]
```diff
@@ -0,0 +1,541 @@
...
+                        }
+                    }
+                }
+            } catch (Exception throwables) {
+                LOGGER.error(throwables.getMessage());
+                return new SqlStatement[0]; // If any exception occurs don't do any migration
+            }
+        }
+
+        return statements.toArray(new SqlStatement[0]);
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/filter/server/migrations/MigrateEquipmentFilter.java:291]:
```suggestion
        return statements.toArray();
```

Comment (basseche) [reply src/main/java/org/gridsuite/filter/server/migrations/MigrateEquipmentFilter.java:291]: if you do so you have to do a cast

File: src/main/java/org/gridsuite/filter/server/migrations/MigrateEquipmentFilter.java (hunk 7 above)
Comment (basseche) [reply src/main/java/org/gridsuite/filter/server/migrations/MigrateEquipmentFilter.java:23]: ok

File: pom.xml (hunk 1 above)
Comment (basseche) [reply pom.xml:48]: ok

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 2 above)
Comment (basseche) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:68]: we will do it in another PR as said Christelle and Thibault

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 6 above)
Comment (basseche) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:68]: yes but we will do it in another PR after the migration works and there is no problem we will do another PR where we will clean all the code. Same for commons-ui.

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 4 above)
Comment (basseche) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:70]: deleted. But we will delete the classes in another PR when we remove the criteria filter tables.

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 5 above)
Comment (basseche) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:76]: done

File: pom.xml (hunk 1 above)
Comment (etiennehomer) [reply pom.xml:48]: Finally we keep it like this

**Review by etiennehomer — DISMISSED** [review]

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 9]
```diff
@@ -2148,120 +1409,4 @@ public void testExpertFilterLoadLinkToOtherFilterWithIsPartOfOperator() throws E
         checkFilterEvaluating(expertFilter, expectedResultJson);
     }
 
-    @Test
-    public void testLineFiltersCrudInBatch() throws Exception {
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:2152]: In my opinion, every ancient tests for general features, but using criteria-based filters, should be replaced by equivalent expert filters

[the remark this answers is not in the document]
Comment (basseche) [reply src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:2152]: ok

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 10]
```diff
@@ -220,565 +219,6 @@ public String joinWithComma(Object... array) {
         return join(array, ",");
     }
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:221]: Remove all ununsed methods

**Review by etiennehomer — CHANGES_REQUESTED** [review]

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**